### ISSUE TITLE
Crash with 7z basis sets

### ISSUE THREAD

**Issue by susilehtola** [pyscf/pyscf#685]
I'm trying to run a calculation with the benchmark quality 7ZaPa basis set from the basis set exchange, but PySCF just segfaults:
```
******** <class 'pyscf.scf.hf.RHF'> ********
method = RHF
initial guess = minao
damping factor = 0
level_shift factor = 0
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
SCF conv_tol = 1e-09
SCF conv_tol_grad = 1e-07
SCF max_cycles = 50
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /tmp/7ztest/tmpfnkfzb59
max_memory 4000 MB (current use 68 MB)
Segmentation fault (core dumped)
```
The input is the following
```
from pyscf import gto
from pyscf import dft, scf
import numpy
import os

mol = gto.M(
    verbose = 4,
    symmetry = 0,
    spin = 0,
    atom = '''
Ne 0.0 0.0 0.0
''',
    basis = {
        'Ne' : gto.basis.load('7zapa-nr.txt', 'Ne'),
    }
)

method = scf.RHF(mol)
method.kernel()
```
I uploaded the basis here:
[7zapa-nr.txt](https://github.com/pyscf/pyscf/files/5098228/7zapa-nr.txt)

I've identified the issue to be the l=7 K function in the input. If one removes the K function, the calculation runs and converges without problems.

Does something in PySCF or libcint/qcint have a hardcoded angular momentum limit? Can this be somehow increased?

**Comment by sunqm** [comment]
The integral library does not support 2-electron integrals with basis l >= 7 .
For 1-electron integrals, the limit is l <= 12.

**Comment by susilehtola** [comment]
Ah, okay. What's the bottleneck? Is it the lack of higher-order Rys quadrature roots? Looks like there's a semi-recent paper that shows that the roots can be calculated accurately to even absurdly high values: [Comput. Phys. Comm. 185, 762 (2014)](https://doi.org/10.1016/j.cpc.2013.11.004)

**Comment by sunqm** [comment]
Right, the numerical stability issue for high order rys roots. When I
implemented the integrals, I didn't find a robust way to compute rys roots.
Thanks for the paper. I'll try it in the future release.

On Thu, Aug 20, 2020 at 12:00:00AM -0700, Susi Lehtola wrote:
> Ah, okay. What's the bottleneck? Is it the lack of higher-order Rys quadrature roots? Looks like there's a semi-recent paper that shows that the roots can be calculated accurately to even absurdly high values: [Comput. Phys. Comm. 185, 762 (2014)](https://doi.org/10.1016/j.cpc.2013.11.004)
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/pyscf/pyscf/issues/685#issuecomment-677373489

--

**Comment by susilehtola** [comment]
I just found a 2016 paper from Harry King on Rys roots and weights, too; I haven't read it yet.
https://pubs.acs.org/doi/pdf/10.1021/acs.jpca.6b10004

@sunqm I saw you've implemented (stable?) integrals up to l=14 in libcint, congrats! This will be great for going to larger basis sets. I guess you're working on the PySCF changes?

**Comment by sunqm** [comment]
Code for regular Coulomb integrals are stable. Small tunings for Coulomb
attenuation (the short-range part) may be applied. But generally they work much
better than older version. Updates in pyscf should come soon.

On Thu, Feb 25, 2021 at 10:08:44PM -0800, Susi Lehtola wrote:
> I just found a 2016 paper from Harry King on Rys roots and weights, too; I haven't read it yet.
> https://pubs.acs.org/doi/pdf/10.1021/acs.jpca.6b10004
> 
> @sunqm I saw you've implemented (stable?) integrals up to l=14 in libcint, congrats! This will be great for going to larger basis sets. I guess you're working on the PySCF changes?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/pyscf/pyscf/issues/685#issuecomment-786435269

--

**Comment by susilehtola** [comment]
Awesome!

**Comment by sunqm** [comment]
This is an issue of old pyscf verions. Basis angular momentum up to l=15 were supported by pyscf-2.0 and newer versions.
